services:
  homeassistant:
    image: ghcr.io/home-assistant/home-assistant:stable
    user: "${DOCKER_PUID:-0}:${DOCKER_PGID:-0}"
    privileged: true
    environment:
      - TZ=${DOCKER_TZ:?error}
    devices:
      - ${SKYCONNECT_DEV:-/dev/serial/by-id/usb-Nabu_Casa_SkyConnect_v1.0_0e9226d6be96ed11b33dc698a7669f5d-if00-port0}:/dev/ttyUSB0
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - homeassistant_config:/config
    networks:
      - network
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
      placement:
        constraints:
          - node.role == manager
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.homeassistant.rule=Host(`homeassistant.${TRAEFIK_DOMAIN_NAME:?error}`)"
        - "traefik.http.services.homeassistant.loadbalancer.server.port=8123"

volumes:
  homeassistant_config:
    name: homeassistant_config

networks:
  network:
    name: ${DOCKER_NETWORK:?error}
    external: true
